src/test/kotlin/three/consulting/epoc/repository/ProjectServiceIntegrationTest.kt

```diff
@@ -0,0 +1,112 @@
+package three.consulting.epoc.repository
+
+import org.assertj.core.api.Assertions
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.test.context.ContextConfiguration
+import three.consulting.epoc.IntegrationTest
+import three.consulting.epoc.dto.CustomerDTO
+import three.consulting.epoc.dto.ProjectDTO
+import three.consulting.epoc.service.*
+import java.time.LocalDate
+
+@ContextConfiguration(classes = [ProjectService::class, CustomerService::class])
+class ProjectServiceIntegrationTest : IntegrationTest() {
+
+    @Autowired
+    private lateinit var customerService: CustomerService
+
+    @Autowired
+    private lateinit var projectService: ProjectService
+
+    @Test
+    fun `searching a project for id return a project`() {
+        val project: ProjectDTO = projectService.findProjectForId(1L)!!
+        assertThat(project.name).isEqualTo("test")
+        assertThat(project.description).isEqualTo("testing")
+        assertThat(project.customer!!.id).isEqualTo(1L)
```

Comment:
Does idea let you use null safe cast `project.customer?.id` here? if the double bang fails inside the assertion you get a strange error message, but if you assert that a null is equal to 1 the output is more readable.

Reply:
This is solved by having customer as non-nullable attribute